deep_qa/layers/tuple_matchers/word_token_tuple_matcher.py

```diff
@@ -0,0 +1,60 @@
+from collections import OrderedDict
+from typing import Any, Dict
+
+from ...common.params import pop_choice
+from ..backend import CollapseToBatch, ExpandFromBatch, Squeeze
+from .word_overlap_tuple_matcher import WordOverlapTupleMatcher
+
+
+class WordTokenTupleMatcher:
+    """
+    ``WordTokenTupleMatchers`` operate on tuples with shape ``(batch_size, num_slots, num_words)``.
+    This class gets two inputs of shape ``(batch_size, num_options, num_question_tuples,
+    num_background_tuples, num_slots, num_words)``, collapses them down to the right shape using
+    ``TimeDistributed`` (or equivalent) and passes them through the actual tuple matcher, returning
+    a tensor of tuple matches of shape ``(batch_size, num_options, num_question_tuples,
+    num_background_tuples)``.
+
+    Parameters
+    ----------
+    text_trainer: TextTrainer
+        A reference to the TextTrainer object this TupleMatcher belongs to, so we can call
+        ``embed_input`` with it.
+
+    tuple_matcher_params: Dict[str, Any], default={}
+        Parameters for constructing an underlying TupleMatcher that operates on word token tuples.
+        We only read the "type" key here, which indexes a class in ``word_token_tuple_matchers``,
+        and then pass the rest of the parameters on to that class as ``kwargs``.
+    """
+    def __init__(self, text_trainer, tuple_matcher_params: Dict[str, Any]=None):
+        self.text_trainer = text_trainer
```

Comment:
Ah is this so the api is the same for the other tuple matchers which do embed things?